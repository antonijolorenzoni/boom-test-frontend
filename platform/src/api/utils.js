const HttpStatus = {
  OK: 200,
};

const isStatusSuccess = (status) => status >= 200 && status < 300;
const isStatusOk = (status) => status === HttpStatus.OK;

export { HttpStatus, isStatusSuccess, isStatusOk };
