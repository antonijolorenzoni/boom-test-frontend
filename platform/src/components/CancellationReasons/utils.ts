import { Reason } from 'types/Reason';

export const getActorsAvailable = (reasons: Array<Reason>): Array<string> =>
  Array.from(new Set(reasons.map((r) => r.code.substring(0, 4))));
