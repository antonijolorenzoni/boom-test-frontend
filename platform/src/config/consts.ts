//
// ────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────── I ──────────
//   :::::: A P P L I C A T I O N   C O N S T A N T S   A N D   U T I L S   V A R I A B L E S : :  :   :    :     :        :          :
// ────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────
//

import { OrderStatus } from 'types/OrderStatus';

export const FormErrorKey = {
  REQUIRED: 'forms.error.required',
  PASSWORD_MUST_COINCIDE: 'forms.error.passwordMustCoincide',
  INVALID_PHONE: 'forms.invalidPhone',
};

export const LANGUAGE_LOCAL_MAP: {
  [k: string]: { key: string; momentLocale: string; translation: string; backend: string };
} = {
  ITALIAN: { key: 'it', momentLocale: 'it', translation: 'it', backend: 'ITALIAN' },
  ENGLISH: { key: 'en', momentLocale: 'en', translation: 'en', backend: 'ENGLISH' },
  SPANISH: { key: 'es', momentLocale: 'es', translation: 'es', backend: 'SPANISH' },
  FRENCH: { key: 'fr', momentLocale: 'fr', translation: 'fr', backend: 'FRENCH' },
};

export const LANGUAGE_LOCAL_MAP_PHOTOGRAPHER: {
  [k: string]: { key: string; momentLocale: string; translation: string; backend: string };
} = {
  ITALIAN: { key: 'it', momentLocale: 'it', translation: 'it', backend: 'ITALIAN' },
  ENGLISH: { key: 'en', momentLocale: 'en', translation: 'en', backend: 'ENGLISH' },
  FRENCH: { key: 'fr', momentLocale: 'fr', translation: 'fr', backend: 'FRENCH' },
  SPANISH: { key: 'es', momentLocale: 'es', translation: 'es', backend: 'SPANISH' },
};

export const LANGUAGES = ['en', 'it'];
export const LANGUAGES_LONG: {
  [key: string]: string;
} = {
  ITALIAN: 'ITALIAN',
  ENGLISH: 'ENGLISH',
  FRENCH: 'FRENCH',
  SPANISH: 'SPANISH',
};
export const DEFAULT_LANGUAGE = LANGUAGES[0];

export const SHOOTINGS_DEFAULT_PAGE_SIZE = 500000;
export const SHOOTING_ARCHIVIATION_PERIOD = 30; // days
export const ACCESS_RULES_TYPES = {
  DEFAULT: 'DEFAULT',
  USER: 'USER',
};

export const USER_ROLES = {
  ROLE_ADMIN: 'ROLE_ADMIN',
  ROLE_MANAGER: 'ROLE_MANAGER',
  ROLE_EDITOR: 'ROLE_EDITOR',
  ROLE_OBSERVER: 'ROLE_OBSERVER',
  ROLE_PHOTOGRAPHER: 'ROLE_PHOTOGRAPHER',
  ROLE_CONTACT_CENTER: 'ROLE_CONTACT_CENTER',
  ROLE_SMB: 'ROLE_SMB',
};

export const SHOOTINGS_STATUSES = {
  NEW: 'NEW',
  AUTO_ASSIGNMENT: 'AUTO_ASSIGNMENT',
  PENDING: 'PENDING',
  MATCHED: 'MATCHED',
  ASSIGNED: 'ASSIGNED',
  ACCEPTED: 'ACCEPTED',
  UPLOADED: 'UPLOADED',
  POST_PROCESSING: 'POST_PROCESSING',
  DONE: 'DONE',
  DOWNLOADED: 'DOWNLOADED',
  CANCELED: 'CANCELED',
  RESHOOT: 'RESHOOT',
  ARCHIVED: 'ARCHIVED',
  REFUSED: 'REFUSED',
  UNSCHEDULED: 'UNSCHEDULED',
};

export const PHOTO_TYPES = {
  EVENTS: 'EVENTS',
  FOOD: 'FOOD',
  PRODUCTS: 'PRODUCTS',
  OTHERS: 'OTHERS',
  REAL_ESTATE: 'REAL_ESTATE',
};

export const ORDER_EDITING_FE_STATUSES = {
  PROCESSING: 'PROCESSING',
  DONE: 'DONE',
  ERROR_CREATING: 'ERROR_CREATING',
  CREATING: 'CREATING',
  ERROR_PROCESSING: 'ERROR_PROCESSING',
  ERROR_ZIPPING: 'ERROR_ZIPPING',
  DONE_WITHOUT_ZIP: 'DONE_WITHOUT_ZIP',
};

export const MAP_DATA_PIPELINE_TO_FE_STATUS: {
  [key: string]: string;
} = {
  CREATING: ORDER_EDITING_FE_STATUSES.CREATING,
  CREATED: ORDER_EDITING_FE_STATUSES.CREATING,
  SENDING: ORDER_EDITING_FE_STATUSES.CREATING,
  SENT: ORDER_EDITING_FE_STATUSES.CREATING,
  UPDATED_EDITOR: ORDER_EDITING_FE_STATUSES.CREATING,
  REFUSED: ORDER_EDITING_FE_STATUSES.CREATING,

  PROCESSING: ORDER_EDITING_FE_STATUSES.PROCESSING,
  PROCESSED: ORDER_EDITING_FE_STATUSES.PROCESSING,
  DOWNLOADING: ORDER_EDITING_FE_STATUSES.PROCESSING,

  DOWNLOADED: ORDER_EDITING_FE_STATUSES.DONE_WITHOUT_ZIP,

  ZIPPED: ORDER_EDITING_FE_STATUSES.DONE,
  NOTIFIED: ORDER_EDITING_FE_STATUSES.DONE,
  DONE: ORDER_EDITING_FE_STATUSES.DONE,
  RESHOOT: ORDER_EDITING_FE_STATUSES.DONE,

  FAILED_TO_CREATE: ORDER_EDITING_FE_STATUSES.ERROR_CREATING,
  SEND_FAILED: ORDER_EDITING_FE_STATUSES.ERROR_CREATING,

  PROCESSING_FAILED: ORDER_EDITING_FE_STATUSES.ERROR_PROCESSING,
  DOWNLOAD_FAILED: ORDER_EDITING_FE_STATUSES.ERROR_PROCESSING,
  NOTIFICATION_FAILED: ORDER_EDITING_FE_STATUSES.ERROR_PROCESSING,

  ZIP_FAILED: ORDER_EDITING_FE_STATUSES.ERROR_ZIPPING,
};

export const AUTO_ASSIGNMENT_STATUSES = {
  DEFAULT: 'DEFAULT',
  INVITED: 'INVITED',
  INSERTED: 'INSERTED',
  REFUSED: 'REFUSED',
  ELAPSED: 'ELAPSED',
  FREEZED: 'FREEZED',
};

export const PERMISSIONS = {
  READ: 'READ',
  CREATE: 'CREATE',
  DELETE: 'DELETE',
  UPDATE: 'UPDATE',
  ACL: 'ACL',
  EVALUATE: 'EVALUATE',
  MANAGE: 'MANAGE',
  DOWNLOAD: 'DOWNLOAD',
  UNAVAILABILITIES: 'UNAVAILABILITIES',
  PHOTOGRAPHER: 'PHOTOGRAPHER',
  REPORT: 'REPORT',
  ACCEPT_PHOTOS: 'ACCEPT_PHOTOS',
  REFUSE_PHOTOS: 'REFUSE_PHOTOS',
  UNASSIGN: 'UNASSIGN',
  ASSIGN: 'ASSIGN',
  CANCEL: 'CANCEL',
  BALANCE: 'BALANCE',
};

export const PERMISSION_ENTITIES = {
  ORGANIZATION: 'ORGANIZATION',
  COMPANY: 'COMPANY',
  USER: 'USER',
  PHOTOGRAPHER: 'PHOTOGRAPHER',
  SHOOTING: 'SHOOTING',
  PRICINGPACKAGE: 'PRICINGPACKAGE',
  INVOICEITEM: 'INVOICEITEM',
  CHECKLIST: 'CHECKLIST',
  AUTHORIZEDSERVICE: 'AUTHORIZEDSERVICE',
  COMPANYPENALTIESCONFIG: 'COMPANYPENALTIESCONFIG',
};

export const PHOTOGRAPHER_SHOOTING_CALENDAR_STATUSES = {
  ASSIGNED: 'ASSIGNED',
  ACCEPTED: 'ACCEPTED',
  AUTO_ASSIGNMENT: 'AUTO_ASSIGNMENT',
  UPLOADED: 'UPLOADED',
  POST_PROCESSING: 'POST_PROCESSING',
  DONE: 'DONE',
  DOWNLOADED: 'DOWNLOADED',
  REFUSED: 'REFUSED',
  RESHOOT: 'RESHOOT',
  CANCELED: 'CANCELED',
  NEW_INVITE: 'NEW_INVITE',
};

export const PHOTOGRAPHER_SHOOTING_LIST_STATUSES = {
  DONE: 'DONE',
  RESHOOT: 'RESHOOT',
  DOWNLOADED: 'DOWNLOADED',
  CANCELED: 'CANCELED',
};

export const COMPANIES_SHOOTING_CALENDAR_STATUSES = {
  ACCEPTED: 'ACCEPTED',
  POST_PROCESSING: 'POST_PROCESSING',
  CANCELED: 'CANCELED',
  DONE: 'DONE',
  DOWNLOADED: 'DOWNLOADED',
  RESHOOT: 'RESHOOT',
  ARCHIVED: 'ARCHIVED',
};

export const COMPANIES_SHOOTING_LIST_STATUSES = {
  DONE: 'DONE',
  DOWNLOADED: 'DOWNLOADED',
  ARCHIVED: 'ARCHIVED',
};

export const COMPANIES_ORDER_STATUSES_TRANSLATION_MAP = {
  NEW: 'ACCEPTED',
  PENDING: 'ACCEPTED',
  MATCHED: 'ACCEPTED',
  ASSIGNED: 'ACCEPTED',
  ACCEPTED: 'ACCEPTED',
  UPLOADED: 'ACCEPTED',
  POST_PROCESSING: 'POST_PROCESSING',
  DONE: 'DONE',
  DOWNLOADED: 'DOWNLOADED',
  CANCELED: 'CANCELED',
  RESHOOT: 'RESHOOT',
  ARCHIVED: 'ARCHIVED',
  REFUSED: 'POST_PROCESSING',
  AUTO_ASSIGNMENT: 'ACCEPTED',
  UNSCHEDULED: 'UNSCHEDULED',
};

export const PHOTOGRAPHER_SHOOTING_STATUSES_TRANSLATION_MAP = {
  NEW: 'NEW',
  PENDING: 'NEW_INVITE',
  MATCHED: 'MATCHED',
  ASSIGNED: 'NEW_INVITE',
  ACCEPTED: 'ACCEPTED',
  UPLOADED: 'UPLOADED',
  POST_PROCESSING: 'POST_PROCESSING',
  DONE: 'DONE',
  DOWNLOADED: 'DONE',
  CANCELED: 'CANCELED',
  RESHOOT: 'RESHOOT',
  ARCHIVED: 'ARCHIVED',
  REFUSED: 'REFUSED',
  AUTO_ASSIGNMENT: 'NEW_INVITE',
};

export const HIDDEN_DELETE_BUTTON_STATUSES = {
  CANCELED: 'CANCELED',
  ARCHIVED: 'ARCHIVED',
  RESHOOT: 'RESHOOT',
  DONE: 'DONE',
  DOWNLOADED: 'DOWNLOADED',
  POST_PROCESSING: 'POST_PROCESSING',
};

export const TERMS_AND_CONDITIONS_URL = 'https://boom.co/terms-conditions';
export const PRIVACY_POLICY_URL = 'https://boom.co/privacy-policy';

export const INVOICE_ITEMS_TYPES = {
  PHOTOGRAPHER_REFUND: 'PHOTOGRAPHER_REFUND',
  REFUSE_PENALTY: 'REFUSE_PENALTY',
  RESHOOT_PENALTY: 'RESHOOT_PENALTY',
  SHOOTING_CANCELED: 'SHOOTING_CANCELED',
  SHOOTING_RESCHEDULED: 'SHOOTING_RESCHEDULED',
  SHOOTING_COMPANY_PRICE: 'SHOOTING_COMPANY_PRICE',
  SHOOTING_PHOTOGRAPHER_EARNING: 'SHOOTING_PHOTOGRAPHER_EARNING',
  PHOTOGRAPHER_PENALTY: 'PHOTOGRAPHER_PENALTY',
  COMPANY_REFUND: 'COMPANY_REFUND',
  COMPANY_PENALTY: 'COMPANY_PENALTY',
  SHOOTING_PHOTOGRAPHER_REFUND: 'SHOOTING_PHOTOGRAPHER_REFUND',
  SHOOTING_TRAVEL_EXPENSES: 'SHOOTING_TRAVEL_EXPENSES',
  PHOTOGRAPHER_TRAVEL_EXPENSES: 'PHOTOGRAPHER_TRAVEL_EXPENSES',
  SHOOTING_REFUNDED_TRAVEL_EXPENSES: 'SHOOTING_REFUNDED_TRAVEL_EXPENSES',
  PHOTOGRAPHER_TRAVEL_REFUND: 'PHOTOGRAPHER_TRAVEL_REFUND',
  COMPANY_TRAVEL_PENALTY: 'COMPANY_TRAVEL_PENALTY',
  PHOTOGRAPHER_REFUNDED_TRAVEL_EXPENSES: 'PHOTOGRAPHER_REFUNDED_TRAVEL_EXPENSES',
};

export const INVOICE_ITEMS_OPTIONS_TYPES = {
  PHOTOGRAPHER_TRAVEL_EXPENSES: 'PHOTOGRAPHER_TRAVEL_EXPENSES',
  PHOTOGRAPHER_PENALTY: 'PHOTOGRAPHER_PENALTY',
  COMPANY_REFUND: 'COMPANY_REFUND',
  COMPANY_PENALTY: 'COMPANY_PENALTY',
};

export const PENALTIES_TYPES = {
  NO_PENALTY: 'NO_PENALTY',
  HALF_PENALTY: 'HALF_PENALTY',
  FULL_PENALTY: 'FULL_PENALTY',
};

export const SEVERITY_TYPES = {
  FULL: 'FULL',
  HALF: 'HALF',
  UNCLASSIFIED: 'UNCLASSIFIED',
};

export const REFUND_TYPES = {
  NO_REFUND: 'NO_REFUND',
  HALF_REFUND: 'HALF_REFUND',
  FULL_REFUND: 'FULL_REFUND',
};

export const DEFAULT_CURRENCY = {
  id: 1,
  alphabeticCode: 'EUR',
  displayName: 'Euro',
  numericCode: 978,
  symbol: '€',
};

export const HIDDEN_INVOICE_TYPES_EVENTS = {
  SHOOTING_COMPANY_PRICE: 'SHOOTING_COMPANY_PRICE',
  SHOOTING_PHOTOGRAPHER_EARNING: 'SHOOTING_PHOTOGRAPHER_EARNING',
};

export const HIDDEN_EVEN_TYPE_BOOM = {
  COMPANY_SHOOTING_CANCELLATION_PENALTY: 'COMPANY_SHOOTING_CANCELLATION_PENALTY',
  COMPANY_RESCHEDULE_PENALTY: 'COMPANY_RESCHEDULE_PENALTY',
  COMPANY_SHOOTING_REFUND: 'COMPANY_SHOOTING_REFUND',
  PHOTOGRAPHER_REFUND: 'PHOTOGRAPHER_REFUND',
  SHOOTING_UPLOAD_REMINDER: 'SHOOTING_UPLOAD_REMINDER',
};

export const UNDELETABLE_INVOICE_ITEMS_TYPES = {
  SHOOTING_REFUNDED_TRAVEL_EXPENSES: 'SHOOTING_REFUNDED_TRAVEL_EXPENSES',
  REFUSE_PENALTY: 'REFUSE_PENALTY',
  SHOOTING_TRAVEL_EXPENSES: 'SHOOTING_TRAVEL_EXPENSES',
  SHOOTING_COMPANY_PRICE: 'SHOOTING_COMPANY_PRICE',
  SHOOTING_PHOTOGRAPHER_EARNING: 'SHOOTING_PHOTOGRAPHER_EARNING',
  SHOOTING_RESCHEDULED: 'SHOOTING_RESCHEDULED',
};

export const PHOTOGRAPHER_INVOICE_ITEMS_REFUNDS = {
  PHOTOGRAPHER_PENALTY: 'PHOTOGRAPHER_PENALTY',
  SHOOTING_TRAVEL_EXPENSES: 'SHOOTING_TRAVEL_EXPENSES',
  PHOTOGRAPHER_TRAVEL_EXPENSES: 'PHOTOGRAPHER_TRAVEL_EXPENSES',
};

export const DELIVERY_METHOD_TYPE = {
  EMAIL: 'EMAIL',
  DRIVE: 'DRIVE',
};

export const CUSTOM_PERCENTAGE_CANCELLATION = [
  '0',
  '5',
  '10',
  '15',
  '20',
  '25',
  '30',
  '35',
  '40',
  '45',
  '50',
  '55',
  '60',
  '65',
  '70',
  '75',
  '80',
  '85',
  '90',
  '95',
  '100',
];

export const COMPANY_PENALTIES_HOURS = [1, 2, 3, 4, 5, 6, 7, 8, 9, 10, 11, 12, 13, 14, 15, 16, 17, 18, 19, 20, 21, 22, 23, 24];

export const COMPANY_PENALTIES_PERCENTAGES = [0, 5, 10, 15, 20, 25, 30, 35, 40, 45, 50, 55, 60, 65, 70, 75, 80, 85, 90, 95, 100];

export const SHOOTING_STATES_WIDER_VIEW = ['MATCHED'];

export const SHOOTING_STATUSES_UI_ELEMENTS: { [k: string]: { color: string } | { color: string; secondaryColor: string } | undefined } = {
  [SHOOTINGS_STATUSES.NEW]: {
    color: '#311B92',
  },
  [SHOOTINGS_STATUSES.CANCELED]: {
    color: '#9932CC',
  },
  [SHOOTINGS_STATUSES.REFUSED]: {
    color: '#8b0000',
  },
  [SHOOTINGS_STATUSES.AUTO_ASSIGNMENT]: {
    color: '#7A4AAC',
    secondaryColor: '#7A4AAC80',
  },
  [SHOOTINGS_STATUSES.MATCHED]: {
    color: '#B8860B',
  },
  [SHOOTINGS_STATUSES.PENDING]: {
    color: '#D84315',
  },
  [SHOOTINGS_STATUSES.ASSIGNED]: {
    color: '#FFA500',
    secondaryColor: '#FFA50080',
  },
  [SHOOTINGS_STATUSES.ACCEPTED]: {
    color: '#3F80DB',
  },
  [SHOOTINGS_STATUSES.UPLOADED]: {
    color: '#01579B',
  },
  [SHOOTINGS_STATUSES.POST_PROCESSING]: {
    color: '#E65100',
  },
  [SHOOTINGS_STATUSES.RESHOOT]: {
    color: '#cc0033',
  },
  [SHOOTINGS_STATUSES.DONE]: {
    color: '#b2df8a',
  },
  [SHOOTINGS_STATUSES.DOWNLOADED]: {
    color: '#1C5D1F',
  },
  [SHOOTINGS_STATUSES.ARCHIVED]: {
    color: '#cc0033',
  },
  [SHOOTINGS_STATUSES.UNSCHEDULED]: {
    color: '#311A91',
  },
  [PHOTOGRAPHER_SHOOTING_CALENDAR_STATUSES.NEW_INVITE]: {
    color: '#FFA500',
    secondaryColor: '#FFA50080',
  },
};

export const DELIVERY_STATUS = {
  WAITING: 'WAITING',
  DELIVERED: 'DELIVERED',
  FAILED: 'FAILED',
};

export const NOT_AVAILABLE = 'N/A';

export const KNOWN_VALUE = 'KNOWN';
export const UNKNOWN_VALUE = 'UNKNOWN';

export const CancellationActors = {
  BOOM: 'BOO-',
  CLIENT: 'CLI-',
  BUSINESS_OWNER: 'BOW-',
  PHOTOGRAPHER: 'PHO-',
};

export const ReasonRoles = {
  ADMINISTRATOR: 'Administrator',
  CONTACT_CENTER: 'ContactCenter',
  CLIENT: 'Client',
  PHOTOGRAPHER: 'Photographer',
};

export const UNASSIGNED = 'UNASSIGNED';

export const PHOTO_TYPES_OPEN_DATE = ['REAL_ESTATE', 'FOOD'];

export const CONTACT_CENTER_SLUG = 'contact-center';

export const INTERNAL_EDITING_VALUE = 'INTERNAL';
export const EXTERNAL_EDITING_VALUE = 'EXTERNAL';

export const OrganizationTier = {
  Enterprise: 'enterprise',
  SMB: 'smb',
};

export const AutoAssignmentJobStatus = {
  Running: 'RUNNING',
};

export const SMB_SUBSCRIPTIONS_PRICE = {
  MONTHLY: 29,
};

export const PLACE_HOLDER = '-';

export const MAIL = {
  SALES: 'sales@boom.co',
};

export const publicRoutes = ['/login', '/setPassword', '/insertResetPassword', '/changeMail'];

export const clientStatuses = [
  OrderStatus.Accepted,
  OrderStatus.PostProcessing,
  OrderStatus.Done,
  OrderStatus.Downloaded,
  OrderStatus.Canceled,
  OrderStatus.Reshoot,
  OrderStatus.Archived,
  OrderStatus.Unscheduled,
];

export const rawExtensions = [
  'DNG',
  'CR3',
  'NEF',
  'ARW',
  'CRW',
  'CR2',
  'RAW',
  'ERF',
  'RAF',
  'GPR',
  '3FR',
  'FFF',
  'DCR',
  'KDC',
  'MRW',
  'MOS',
  'IIQ',
  'RWL',
  'MEF',
  'MFW',
  'NRW',
  'ORF',
  'RW2',
  'SRW',
  'PEF',
  'SRF',
];
