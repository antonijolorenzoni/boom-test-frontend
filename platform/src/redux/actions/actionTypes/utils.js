//
// ────────────────────────────────────────────────────────────────────────────────────────────── I ──────────
//   :::::: P L A T F O R M   U T I L S   A C T I O N   T Y P E S : :  :   :    :     :        :          :
// ────────────────────────────────────────────────────────────────────────────────────────────────────────
//

module.exports = {
  SET_SPINNER_VISIBILITY: 'SET_SPINNER_VISIBILITY',
  SET_SELECTED_LANGUAGE: 'SET_SELECTED_LANGUAGE',
  SET_IS_APP_MOBILE: 'SET_IS_APP_MOBILE',
  SET_LOADING_PROGRESS_VISIBLE: 'SET_LOADING_PROGRESS_VISIBLE',
  SET_LOADING_PROGRESS_VALUE: 'SET_LOADING_PROGRESS_VALUE',
  SAVE_PLATFORM_CURRENCIES: 'SAVE_PLATFORM_CURRENCIES',
  SAVE_PLATFORM_CURRENCIES_PAGINATION: 'SAVE_PLATFORM_CURRENCIES_PAGINATION',
  SET_PLATFORM_CURRENCIES_FILTER: 'SET_PLATFORM_CURRENCIES_FILTER',
  RESET_PLATFORM_CURRENCIES_FILTERS: 'RESET_PLATFORM_CURRENCIES_FILTERS',
  SET_CALENDAR_TIMEZONE: 'SET_CALENDAR_TIMEZONE',
};
