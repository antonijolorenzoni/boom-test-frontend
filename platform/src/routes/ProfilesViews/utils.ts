import React from 'react';

interface Empty {}

function isEmpty<A>(item: A | Empty): item is Empty {
  return !Object.entries(item as Empty).length;
}

const Nullable = <A>(E: React.FC<A>) => (props: A | Empty) => (isEmpty(props) ? null : E(props));

export { isEmpty, Nullable };
