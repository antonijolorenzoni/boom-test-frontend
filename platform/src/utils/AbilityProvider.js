//
// ────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────── I ──────────
//   :::::: C O M P A N Y   A N D   O R G A N I Z A T I O N   A B I L I T Y   P R O V I D E R : :  :   :    :     :        :          :
// ────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────
//

import AbilityHelper from './abilityhelpers/AbilityHelper';

class AbilityProvider {
  organizationAbilityHelper = new AbilityHelper(); // For the outer level

  companyAbilityHelper = new AbilityHelper(); // For inner companies

  getOrganizationAbilityHelper() {
    return this.organizationAbilityHelper;
  }

  getCompanyAbilityHelper() {
    return this.companyAbilityHelper;
  }
}

export default new AbilityProvider();
