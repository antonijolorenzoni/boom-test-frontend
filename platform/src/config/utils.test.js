import { revertToOriginalStatuses } from './utils';
import { SHOOTINGS_STATUSES, COMPANIES_ORDER_STATUSES_TRANSLATION_MAP } from './consts';

describe('revertToOriginalStatuses', () => {
  it('should return original statuses from COMPANIES_ORDER_STATUSES_TRANSLATION_MAP for ACCEPTED status', () => {
    expect(revertToOriginalStatuses([SHOOTINGS_STATUSES.ACCEPTED], COMPANIES_ORDER_STATUSES_TRANSLATION_MAP)).toMatchSnapshot();
  });

  it('should return original statuses from COMPANIES_ORDER_STATUSES_TRANSLATION_MAP for POST_PROCESSING status', () => {
    expect(revertToOriginalStatuses([SHOOTINGS_STATUSES.POST_PROCESSING], COMPANIES_ORDER_STATUSES_TRANSLATION_MAP)).toMatchSnapshot();
  });

  it('should return original statuses from COMPANIES_ORDER_STATUSES_TRANSLATION_MAP for a mix of statuses', () => {
    expect(
      revertToOriginalStatuses(
        [SHOOTINGS_STATUSES.DONE, SHOOTINGS_STATUSES.ACCEPTED, SHOOTINGS_STATUSES.POST_PROCESSING],
        COMPANIES_ORDER_STATUSES_TRANSLATION_MAP
      )
    ).toMatchSnapshot();
  });
});
