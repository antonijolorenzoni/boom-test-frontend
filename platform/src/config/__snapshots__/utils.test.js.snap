// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`revertToOriginalStatuses should return original statuses from COMPANIES_ORDER_STATUSES_TRANSLATION_MAP for ACCEPTED status 1`] = `
Array [
  "NEW",
  "PENDING",
  "MATCHED",
  "ASSIGNED",
  "ACCEPTED",
  "UPLOADED",
  "AUTO_ASSIGNMENT",
]
`;

exports[`revertToOriginalStatuses should return original statuses from COMPANIES_ORDER_STATUSES_TRANSLATION_MAP for POST_PROCESSING status 1`] = `
Array [
  "POST_PROCESSING",
  "REFUSED",
]
`;

exports[`revertToOriginalStatuses should return original statuses from COMPANIES_ORDER_STATUSES_TRANSLATION_MAP for a mix of statuses 1`] = `
Array [
  "DONE",
  "NEW",
  "PENDING",
  "MATCHED",
  "ASSIGNED",
  "ACCEPTED",
  "UPLOADED",
  "AUTO_ASSIGNMENT",
  "POST_PROCESSING",
  "REFUSED",
]
`;
